### DISCUSSION TITLE
STI-Algorithms used in SEPIA = STISuite results ?

### DISCUSSION THREAD

**Discussion by zoelir729** [kschan0214/sepia#23] [General]
I have read the closed issue (SEPIA results != STISuite results #4) similar to this one.
However, I haven't found a solution and reason like him. 
Below were my steps. 【single echo】
![image](https://user-images.githubusercontent.com/36666968/126645197-c926f495-c526-4656-927e-651b788b74ee.png)

Left was the results obtained by using Sepia (choose STISuite algorithms). The right was performed in STISuite v3.0.
The two results were very similar, but I found the Right (STISuite) one was smoother and the left one showed a more coarse appearance.
![image](https://user-images.githubusercontent.com/36666968/126646967-404c58f3-99b4-40ab-82ae-c43e6c99b1d0.png)
![image](https://user-images.githubusercontent.com/36666968/126648114-a550d478-d3b9-40fe-95f5-f4cd0c3aaa77.png)

The Infos showed a difference in Intensity Range and others.
![image](https://user-images.githubusercontent.com/36666968/126648466-642f81c5-2134-4254-a3d6-f00b291ca6a2.png)

Is there a reason or an error in my operation? （Plus: In terms of the difference in intensity, I think the masks used were not the main reason.）

**Comment by kschan0214** [comment]
Hi Zoe,

The closed issue #4 you referred is related to the user defined pipeline (involved data transferring between python and matlab) instead of the issue on the SEPIA's side. So I didn't take a second look on the STI suite implementation in SEPIA as the user reported the final results are identical after fixing his issue.

As SEPIA using the methods from the toolbox directly, I would expect given the same input images/maps, the resulting maps would also be the same. However, there are some intermediate steps in SEPIA that may not presented in the STI suite pipeline. These differences could also make impacts on the final QSM map. 

I will try to test this and coming back to you. In the meantime, if you would like to investigate where the differences coming from, I suggest you breaking down the full pipeline into separate steps. By using the same input images (including the signal mask) on each step for both SEPIA and STI suite, and also switch off the utility function (change the 'Removal residual B1' to 'None'), this could give you a better idea where the differences coming from.

P.S. It seems to me you are running QSM processing on high-passed filtered phase. As some of the phase information is lost after the high-pass filtered processing, please be cautious if you intend to use the resulting map for quantitative analysis.

**Comment by zoelir729** [comment]
For high-passed filtered phase. Sorry for lacking the basics of SWI. You mean **SMV size (mm)** ? 12 is large?
![image](https://user-images.githubusercontent.com/36666968/127708051-31a64eb3-28ca-4582-acdc-86266c245097.png)

And if I want to perform VSHARP algorithm like this: The variable-kernels sophisticated harmonic artifact reduction for phase data (V-SHARP) algorithm was used for background field removal with spherical kernel radii from 0.75 to 30 mm/maximum radii between 0.75 and 30 mm.
‘VSHARP’ could be set like the below pic.
![image](https://user-images.githubusercontent.com/36666968/127708222-1859d8d1-175c-4b49-99b1-84673f7405cb.png)
How to set the same parameter in 'VSHARP (STI suite)' ？

**Comment by kschan0214** [comment]
Hi Zoe,

In my test below I couldn't find any significant differences between the results from STI Suite and SEPIA. You can try the attached script as a template for the test on your data.

Left panel: results from STI Suite; right panel: differences between STI suite script vs STI suite in SEPIA
![STISuite_vs_SEPIA](https://user-images.githubusercontent.com/39571502/128132837-cce3d040-923c-404c-9148-31d91dc13c45.png)

[sepia_vs_STIsuite_template.m.zip](https://github.com/kschan0214/sepia/files/6929092/sepia_vs_STIsuite_template.m.zip)

**Comment by kschan0214** [comment]
The attached image (assumed to be QSM map) is lack of strong susceptibility contrast in the basal ganglia (see below, red arrows). This is usually an indicator suggesting the input phase image is already high-pass filtered. (For more typical QSM maps you can check on Figure 1 or 5 in https://doi.org/10.1016/j.neuroimage.2020.117611. 

![126646967-404c58f3-99b4-40ab-82ae-c43e6c99b1d0](https://user-images.githubusercontent.com/39571502/128133741-fca7d37a-4ca7-416d-8a59-53674ad81bd2.png)
